pipeline-scripts/buildlib.groovy

```diff
@@ -873,6 +873,66 @@ def get_distgit_notify( record_log ) {
 
 }
 
+def notify_dockerfile_reconciliations(doozerWorking, sourceBranches) {
+    // loop through all new commits that affect dockerfiles and notify their owners
+
+    record_log = parse_record_log(DOOZER_WORKING)
+    distgit_notify = get_distgit_notify(record_log)
+    distgit_notify = mapToList(distgit_notify)
+
+    for (i = 0; i < distgit_notify.size(); i++) {
+        distgit = distgit_notify[i][0]
+        val = distgit_notify[i][1]
+
+        try {
+            alias = val['source_alias']
+            dockerfile_url = ""
+            github_url = GITHUB_URLS[alias]
+            github_url = github_url.replace(".git", "")
+            github_url = github_url.replace("git@", "")
+            github_url = github_url.replaceFirst(":", "/")
+            dockerfile_sub_path = val['source_dockerfile_subpath']
+            dockerfile_url = "Upstream source file: https://" + github_url + "/blob/" + sourceBranches[alias] + "/" + dockerfile_sub_path
+            // always mail success list, val.owners will be comma delimited or empty
+            commonlib.email(
+                to: "aos-team-art@redhat.com,${val.owners}",
+                from: "aos-cicd@redhat.com",
+                subject: "${val.image} Dockerfile reconciliation for OCP v${params.BUILD_VERSION}",
+                body: """
+Why am I receiving this?
```

Comment:
Is there a way in jenkins groovy to have a text constants file? I mean... we could just have textconsts.groovy and import it inside buildlib.groovy I guess. Multi-line strings in groovy are terrible.
@sosiouxme I realize this is not your fault... it was like this before. Just pontificating.

Reply:
yes, but using a resource bundle introduces complications too. typically there are things you want to format in. then you have to move back and forth between the locations when you want to change anything. I feel that embedding the text in/near the code is the lesser evil. may improve things if we extract the text to a constant right before the method using it, and use formatting instead of interpolation.